src/main/java/thrift/logic/parser/UntagCommandParser.java

```diff
@@ -28,21 +27,20 @@ public UntagCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_INDEX, CliSyntax.PREFIX_TAG);
 
         Index index;
-        Set<Tag> tagSet = new HashSet<Tag>();
+        Set<Tag> tagSet;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreambleIncludeIndex());
-            for (String tagName : argMultimap.getAllValues(CliSyntax.PREFIX_TAG)) {
-                if (!tagName.isEmpty()) {
-                    Tag tag = new Tag(tagName);
-                    tagSet.add(tag);
-                }
+            String indexStr = argMultimap.getSingleValue(CliSyntax.PREFIX_INDEX)
+                    .orElseThrow(() -> new ParseException(""));
+            index = ParserUtil.parseIndex(indexStr);
+            tagSet = ParserUtil.parseTags(argMultimap.getAllValues(CliSyntax.PREFIX_TAG));
+            if (tagSet.isEmpty()) {
+                throw new ParseException(UntagCommand.MESSAGE_NOT_UNTAGGED);
             }
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UntagCommand.MESSAGE_USAGE), pe);
-        } catch (IllegalArgumentException iae) {
```

Comment:
Is it alright to remove the catch block for `IllegalArgumentException` here? If it never catches anything in your latest version disregard this comment.

Reply:
The block (for-loop for tagSet) that throws the `IllegalArgumentException` has been removed from the code so it doesn't throw it anymore! 👍